controllers/reconcilers/prometheus_installation/prometheus_installation_reconciler.go

```diff
@@ -219,3 +233,54 @@ func (r *Reconciler) reconcileOperatorgroup(ctx context.Context, cr *v1.Observab
 
 	return v1.ResultSuccess, nil
 }
+
+func (r *Reconciler) removePrometheusOperatorIndexResources(ctx context.Context, source *v1alpha1.CatalogSource, cr *v1.Observability) error {
+	subscription := &v1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "prometheus-subscription",
+			Namespace: cr.GetPrometheusOperatorNamespace(),
+		},
+	}
+	err := r.client.Delete(ctx, subscription)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	// Delete catalog source
+	err = r.client.Delete(ctx, source)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	// We have to remove the prometheus operator deployment manually
+	deployments := &appsv1.DeploymentList{}
+	opts := &client.ListOptions{
+		Namespace: cr.Namespace,
+	}
+	err = r.client.List(ctx, deployments, opts)
+	if err != nil {
+		return err
+	}
+
+	for _, deployment := range deployments.Items {
```

Comment:
will deleting the subscription not also delete the deployment?

Reply:
I used the same pattern here as [Cleanup](https://github.com/redhat-developer/observability-operator/blob/main/controllers/reconcilers/prometheus_installation/prometheus_installation_reconciler.go#L33) stage. I can recheck this and see if the deployment check is unnecessary

Reply:
Deployment delete step not needed and removed. Also removed from general Cleanup stage. Thanks @pb82